### PR TITLE
Archery Redux Part 1 (Arrows/Bolts)

### PR DESCRIPTION
**Pull request by darktoes**
First part of my archery redux, focus on making arrows/bolts sensible so they don't magically make crappy bows/crossbows vastly more powerful.

#### Summary
```SUMMARY: Balance "Make arrows/bolts more reasonable"```  

#### Purpose of change
Archery is all over the place at the moment. I'm trying to fix that, starting with the fact that arrows and bolts are weird, and high-quality arrows cause disproportionately high damage when used in a low power bow. Use of prop_damage should fix this nicely, though I'm still waiting on someone to resolve my prop_pierce issue.

#### Describe the solution
Basically all arrows and bolts are more reasonable, with emphasis on realism. Generally speaking, arrows use proportional damage rather than flatly adding damage, so that top quality arrows in a bottom quality bow don't make for unreasonably high damage.

Additionally, a better naming convention will make it much easier for players to recognise which arrows are superior, and what they should work toward to improve their damage output.

#### Additional context
Changes and their rationale are outlined here:
https://docs.google.com/spreadsheets/d/1wpuAk7F1fJ227yAgYbokwn7KxdEeOv-MGFvnnnW5s1Q/edit#gid=244216825

### CODE REVIEW

File: data/json/items/archery.json [hunk 1]
```diff
@@ -2,258 +2,247 @@
...
-    "damage": 6,
-    "pierce": 3,
-    "range": 7,
-    "dispersion": 135,
+    "prop_damage": 0.5,
+    "dispersion": 130,
     "loudness": 0,
     "count": 10,
-    "effects": [ "RECOVER_10" ]
+    "effects": [ "RECOVER_35" ]
```
Comment (MetroidHunter) [inline data/json/items/archery.json]: If this is a game arrow, does it need NOGIB here?

File: data/json/items/archery.json [hunk 2]
```diff
@@ -2,258 +2,247 @@
...
     "ammo_type": "arrow",
-    "damage": 3,
-    "range": 6,
+    "prop_damage": 0.5,
+    "range": -6,
     "dispersion": 210,
     "loudness": 0,
     "count": 10,
-    "effects": [ "RECOVER_5", "NOGIB" ]
+    "effects": [ "RECOVER_10", "NOGIB" ]
```
Comment (MetroidHunter) [inline data/json/items/archery.json]: Now that this is not a game arrow, does this no longer need NOGIB?

File: data/json/items/archery.json (hunk 1 above)
Comment (darktoes) [reply data/json/items/archery.json]: It does, thank you for the save.

File: data/json/items/archery.json (hunk 2 above)
Comment (darktoes) [reply data/json/items/archery.json]: While it isn't explicitly a small game arrow, the blunt tip serves the same purpose. The description reflects this, so hopefully players won't get confused.

That said, a constant naming convention would be a good idea.

**Comment by darktoes** [comment]
Glad you're looking forward to it, and I really appreciate the input.

- Renamed the tipped arrow to "makeshift wooden arrow". The idea is that it's a wooden shaft with a makeshift (non-forged-steel) arrowhead, made of bone, ceramic, or something like that.

- I tried to adjust the naming convention a bit, but the idea is that there are three main "quality" levels of arrow, simple wooden, wooden, and aluminum. I tried to make it "[quality] [type] arrow" but I'm still open to suggestions.

- Aluminum arrows are going to be become uncraftable. Most modern arrows use field points or broadheads, so the player is only realistically going to find aluminium shafts with field points and not bodkins, while bodkins are somewhat easier to make and are historical, so found/crafted wooden arrows have bodkin points. Field points are very similar to bodkin points, and perform identically ingame.

- Blunt simple wooden arrow is basically a crappy small game arrow. I renamed it to simple small game arrow to try to keep with convention.

**Comment by AMurkin** [comment]
Remove "The format is: " before the summary to fix Invalid Summary line.

**Comment by darktoes** [comment]
Whoops! Never missed that one before.

**Comment by darktoes** [comment]
jenkins rebuild

**Comment by MetroidHunter** [comment]
> Glad you're looking forward to it, and I really appreciate the input.
> 
> * Renamed the tipped arrow to "makeshift wooden arrow". The idea is that it's a wooden shaft with a makeshift (non-forged-steel) arrowhead, made of bone, ceramic, or something like that.
> * I tried to adjust the naming convention a bit, but the idea is that there are three main "quality" levels of arrow, simple wooden, wooden, and aluminum. I tried to make it "[quality] [type] arrow" but I'm still open to suggestions.

Ah i see now. Perhaps they could be `simple arrow`, `wooden arrow`, and `aluminum arrow` for clarity in that case. Now that I know `simple wooden` is meant as a quality bar, I can understand the distinctions now but since it wasnt immediately apparent to me, Im worried it wont be to others. 
> * Aluminum arrows are going to become uncraftable. Most modern arrows use field points or broadheads, so the player is only realistically going to find aluminium shafts with field points and not bodkins, while bodkins are somewhat easier to make and are historical, so found/crafted wooden arrows have bodkin points. Field points are very similar to bodkin points, and perform identically ingame.

I dont think field points and bodkins perform the same in real life though. Perhaps from an accuracy standpoint but the longer and sharper point of the bodkin should have a higher grain and piercing ability. Despite that, I can get behind letting them be the same in game for ease. As far as finding them, bodkin points could still be made by hand at the forge but perhaps the recipe should be from a book.

I see in the spreadsheet that you mention the non-craftable aluminum arrows. Is there a game reason to do that or is this more realism? I think by making aluminum ingots from aluminum cans and using a sand mold, a fairly usable aluminum arrow could be made with just the crucible and tongs. Also, the `swage and die` set in the game would probably have something usable for a pipe extruder. Perhaps aluminum arrows should remain craftable and the uncraftable arrows you mention should instead be the carbon fiber arrows?
> * Blunt simple wooden arrow is basically a crappy small game arrow. I renamed it to simple wooden small game arrow to try to keep with convention.

**Comment by darktoes** [comment]
Dropping the "wooden" from "simple wooden arrow" isn't a bad idea, but I think I'd like to keep the main material in the name for now. If people are confused later on I'll look into changing it. If nothing else looking at the actual stats in the crafting menu/inventory should set people right easily enough.

Field points and bodkins naturally aren't going to perform identically in real life, but they're close enough that I don't see a reason to make a distinction, and I'd like to make arrows spawn around the place a lot more, so realism there would be important.

Personally, I think a small pipe extruder jig should be easy enough for the player to make, but I've talked back and forth about it with Kevin a fair bit and I think the plan is to wait until a lathe or more elaborate crafting tools are added so the player can make an accurate extruder jig that way, but if you can convince Kevin otherwise I'd be more than happy to make them craftable.

**Comment by SirPendrak** [comment]
"type": "AMMO",
    "id": "bolt_wood_small_game",
    "price": 1000,
    "name": "metal crossbow bolt",
    "name": "wooden small game arrow", 

"Arrow" is probably a typo. 
Overall changes look good, Powerfull bows and crossbows with shit ammo wont be as effective as used to be. I have only one question about proportional damage. Lets say we shoot the most powerfull crossbow -> damage 50, once with broadhead bolt, and once with bodkin, at a target wchich has 55 armor. 
Bodkin - 50x1,12 = 56 dmg and 3 pierce. 56 - (55 - 3) = 4dmg dealt
Broadhead - 50x1,25 = 65 dmg and one pierce. 65 - (55 - 1) = 11dmg dealt 
Am i doint my calculations wrong, or will it be favouring just biggest damage-dealing ammunition on strong weapons aganist very armored targets?

**Comment by darktoes** [comment]
Fixed the arrow mentions, sadly the dangers of copy+paste strike again.

I hadn't actually done the math, and that's a sad byproduct of the fact that proportional pierce isn't a thing yet. Once someone handles https://github.com/CleverRaven/Cataclysm-DDA/issues/27085 I'll apply appropriate proportional pierce values to all bolts and arrows and it will work out very nicely regardless of which bow or crossbow is used.

**Comment by darktoes** [comment]
Now that I've finished new recipes for arrows and bolts I'm happy that this is ready for merging (once properly reviewed and feed-backed, of course).

**Comment by MetroidHunter** [comment]
> I hadn't actually done the math, and that's a sad byproduct of the fact that proportional pierce isn't a thing yet. Once someone handles #27085 I'll apply appropriate proportional pierce values to all bolts and arrows and it will work out very nicely regardless of which bow or crossbow is used.

Ive got 2 bugs ive committed to fixing but after that, I can look at handling the pierce aspect you mention in #27085

File: data/json/items/crossbows.json [hunk 3]
```diff
@@ -1,110 +1,225 @@
 [
   {
     "type": "AMMO",
-    "id": "bolt_wood",
```
Comment (MetroidHunter) [inline data/json/items/crossbows.json]: I'm not terribly versed in how the game resolves ids from json into objects but wont changing this id break existing items in the game? I assume thats why you didnt change the ids of the arrows

Comment (darktoes) [reply data/json/items/crossbows.json]: That entry still exists, I just apparently shifted it further down the file and Github didn't pick up on that. Since the id still exists it should be perfectly fine, placement in the file has no effect to my knowledge.

**Comment by darktoes** [comment]
I'd really appreciate it if you sorted that for me, if that gets finished then I can add prop_pierce to all the bolts and arrows and they'll be pretty much feature complete. With the exception of the exceptionally wonky fire/explosive arrows.

File: data/json/items/crossbows.json (hunk 3 above)
Comment (MetroidHunter) [reply data/json/items/crossbows.json]: Those things are always hard to see. Thanks for the clarity

Comment (darktoes) [reply data/json/items/crossbows.json]: The entire file is fairly wonky and there's an argument to be made for shifting the existing IDs to new ones, but that's not a very high priority for me at the moment.

**Comment by Treah** [comment]
A wooden arrow with a steel tip should do more pierce then a aluminum arrow with the same tip. This is due to having more mass from the material. Aluminum is much lighter then wood and would have less penetrating power.

**Comment by darktoes** [comment]
You're correct, but I have a hard time justifying a rarer, more difficult to craft arrow being less damaging than a common, easy to craft one. My assumption is that the found aluminum arrows have been weighted down by the player, or simply that the higher quality of manufacturing offsets the somewhat reduced energy transfer efficiency.

**Comment by MetroidHunter** [comment]
> You're correct, but I have a hard time justifying a rarer, more difficult to craft arrow being less damaging than a common, easy to craft one. My assumption is that the found aluminum arrows have been weighted down by the player, or simply that the higher quality of manufacturing offsets the somewhat reduced energy transfer efficiency.

what if... we had.... wood aluminum arrows >.>

**Comment by darktoes** [comment]
But... Why? How?

**Comment by mding9** [comment]
Isn't aluminum not very dense? Even if it can still provide enough weight to the front of the arrow, it wouldn't be as much as a metal-tipped one, meaning it wouldn't be as strong or accurate. Unless you mean an aluminum arrowhead is something between top heavy arrows and non top heavy arrows, so that it flies farther than a metal ripped arrow and is more accurate than an untipped one.

**Comment by MetroidHunter** [comment]
> Isn't aluminum not very dense? Even if it can still provide enough weight to the front of the arrow, it wouldn't be as much as a metal-tipped one, meaning it wouldn't be as strong or accurate. Unless you mean an aluminum arrowhead is something between top heavy arrows and non top heavy arrows, so that it flies farther than a metal ripped arrow and is more accurate than an untipped one.

TL;DR;

- Aluminum is 3 times as dense as wood
- Wood arrows are usually made of light wood and are typically lighter than aluminum arrows
- Aluminum arrows are considered the heaviest arrows among modern arrows

Interesting Things Below:

I think we may have some social conditioning to think Aluminum is less dense than wood, even I at first thought this. But, doing a bit of googling, while the alloy does matter and what kind of wood, it appears that aluminum is a good bit more dense than wood. 

Here is a random chart showing density of some materials:
http://www.psyclops.com/tools/technotes/materials/density.html

Another showing the same:
https://www.engineeringtoolbox.com/density-solids-d_1265.html
https://www.engineeringtoolbox.com/wood-density-d_40.html

Hers another article that goes ahead and talks directly about weight:
http://www.zenithair.com/kit-data/ht-85-12.html

Now that we have density, we just need the volume of the arrow. We can get the diameter which is well explained here:
https://www.bowhunter-ed.com/michigan/studyGuide/Arrow-Shaft-Length-and-Labeling/301023_3694/

And lets assume a 31inch arrow because why not.

31 inches is 0.7874 meters
20/64 diameter in inches is 0.0079375 in meters
13/1000 thickness in inches is 0.0003302 in meters

A hollow tube volume is calculated by pi * height * (outer radius ^2 - inner radius ^2)
height is 0.7874
outer radius is 0.00396875 (half of the diameter)
inner radius is 0.00363855 (outer radius - thickness)

giving us a volume of 0.0000062137 m^3 (for just the shaft)

A wooden shaft though would only have the volume of a cylinder.
Cylinder volume is calculated by pi * height * radius ^ 2
height is still 0.7874
radius is 0.00386875

giving us a volume of 0.0000370243 m^3

Given that Density = Mass/Volume and Weight = Mass * Gravity where earth gravity is naturally 9.807 m/s^2, we can conclude that if pure aluminum is 2.7 10^3 kg/m^3 and seasoned wood from an oak tree is 600 - 900 kg/m3, and lets take the high end of 900 kg/m3, then we can normalize aluminum to 2700 kg/m^3 and get that a 0.0000062137 m^3 shaft has a mass of 0.01407699 kg or 14.1 grams. This gives us a weight of .138 KGMS/2 (aka Newtons) which is 0.03 lbs for the aluminum shaft. Pretty light! We can then take oak wood at 900 kg/m^3 and get that a 0.0000370243 m^3 shaft has a mass of 33.32 grams. This gives us a weight of .327N which is 0.07 lbs. 

So the wooden arrow is heavier because its a solid piece of wood BUT, we assume a lot here. Most professional arrows are cut from light and strong wood like cedar or pine, not oak. A white pine arrow of the same diameter would be 500 kg/m^3 in density giving us a total weight of 0.04 lbs. 

Also, aluminum arrows can have larger diameters and larger thicknesses, changing their weight. If we instead had an alumnium arrow with 16/1000 thickness and 23/64 diameter, we would get a total weight of 0.05lbs.

To avoid most of this math, arrows are mostly measured in GPI (grains per inch) and each component contributes its grains to the total grain of the arrow. This allows a more normalized system across all the arrows.

I think we may be predisposed to neglect aluminum because of our very thin and common applications of it in aluminum foil and aluminum cans

**Comment by darktoes** [comment]
To add to that, and cover the question directly, the aluminum arrows ingame are a hollow aluminum shaft with a steel tip, so they're very light, and very easy to add weight to. That said, wooden shafts are naturally going to be a bit rougher and have minor wood imperfections, while extruded aluminum is (mostly) homogenous. You get a stronger, lighter, more consistent arrow from aluminum, so that's represented ingame with higher durability and slightly lower dispersion. In addition, the aluminum arrows are commercially made, meaning their general quality level is higher. When they're made craftable, it will be with much more elaborate tool requirements.

**Comment by dissociativity** [comment]
https://github.com/CleverRaven/Cataclysm-DDA/issues/28607
made an issue related to this PR

**Comment by Shodan14** [comment]
Soo will thing be merged?

**Comment by darktoes** [comment]
Some time after I take out the range values on the arrows/bolts so they don't crash the game and possibly fiddle the crafting recipes around a bit more. Currently I've got assignments for all 4 of my classes due on Monday, so that'll get done Soon<sup>TM</sup>.

For the record I fully intend to continue/finish my redux, I'm just currently so bogged down in Uni work I don't have the time.

**Comment by Shodan14** [comment]
> 
> 
> Some time after I take out the range values on the arrows/bolts so they don't crash the game and possibly fiddle the crafting recipes around a bit more. Currently I've got assignments for all 4 of my classes due on Monday, so that'll get done SoonTM.
> 
> For the record I fully intend to continue/finish my redux, I'm just currently so bogged down in Uni work I don't have the time.
Looking forward to trying it out.

**Comment by darktoes** [comment]
Removed the range penalties from bolts/arrows and (hopefully) fixed the file conflict. Not really sure what its issue was TBH. Good chance I screwed something up since it's 2AM but if I didn't, then this is probs ready for merging.

**Comment by darktoes** [comment]
jenkins rebuild

**Comment by darktoes** [comment]
NOW it should be okay. Bad code style was causing Gorgon to fail and nobody likes it when Gorgon fails.

**Comment by darktoes** [comment]
jenkins rebuild

**Comment by darktoes** [comment]
Fixed another random merge conflict. It seems to just hate me.

File: data/json/recipes/recipe_ammo.json [hunk 4]
```diff
@@ -299,77 +268,195 @@
...
-      [ [ "steel_chunk", 1 ], [ "scrap", 3 ] ],
-      [ [ "stick", 2 ], [ "broom", 2 ], [ "2x4", 2 ], [ "bee_sting", 2 ] ],
+      [ [ "steel_tiny", 1, "LIST" ] ],
+      [ [ "stick", 1 ], [ "broom", 1 ], [ "2x4", 1 ], [ "bee_sting", 1 ] ],
       [ [ "fletching", 1, "LIST" ] ]
     ]
   },
   {
     "type": "recipe",
-    "result": "arrow_metal",
```
Comment (dgolembo) [inline data/json/recipes/recipe_ammo.json]: Is it just me or is there no longer a arrow_metal recipe at all?  I have had issues seeing the 'aluminium broadhead' arrows in the game - not available even via debug.

**Comment by darktoes** [comment]
It was intentionally removed until we have the capability to make an aluminum extrusion rig.

File: data/json/recipes/recipe_ammo.json (hunk 4 above)
Comment (darktoes) [reply data/json/recipes/recipe_ammo.json]: It was intentionally removed until we have the capability to make an aluminum extrusion rig.

**Comment by dgolembo** [comment]
So, no metal broadhead arrow is intentional?  Fair enough.

File: data/json/recipes/recipe_ammo.json (hunk 4 above)
Comment (dgolembo) [reply data/json/recipes/recipe_ammo.json]: Would it not be less surprising (at least for existing characters) to leave the metal arrow in until the recipe can be updated with new requirements?  Just wondering.

Comment (ZhilkinSerg) [reply data/json/recipes/recipe_ammo.json]: Removing the recipe doesn't remove resulting item.

Comment (dgolembo) [reply data/json/recipes/recipe_ammo.json]: I realise this, my point is that the recipe that used to be there is not there any longer until some future time - meanwhile the capability that used to be there disappeared without any reason.  Does this make the game more fun?

Comment (darktoes) [reply data/json/recipes/recipe_ammo.json]: 1) It makes very little difference.
2) Argue with Kevin if you care that much.

Comment (ZhilkinSerg) [reply data/json/recipes/recipe_ammo.json]: Just scrap removed recipes into mod if you care.

Comment (darktoes) [reply data/json/recipes/recipe_ammo.json]: Oh? Can you tell me what mod it is? (Purely out of curiosity)

Comment (ZhilkinSerg) [reply data/json/recipes/recipe_ammo.json]: The one you will create :)

Comment (darktoes) [reply data/json/recipes/recipe_ammo.json]: Ah. I misunderstood.

Comment (dgolembo) [reply data/json/recipes/recipe_ammo.json]: How does one go about creating a mod?  Is there a document somewhere?  I'm quite new to the game.

Comment (ZhilkinSerg) [reply data/json/recipes/recipe_ammo.json]: Start here - https://github.com/CleverRaven/Cataclysm-DDA/wiki/Guide-to-adding-new-content-to-CDDA-for-first-time-contributors.
